web/src/app/services/data-store/data-store.service.ts

```diff
@@ -150,26 +167,39 @@ export class DataStoreService {
       )
       .snapshotChanges()
       .pipe(
+        map(this.documentChangeToSurvey),
         map(surveys =>
-          List(
-            surveys
-              .map(a => {
-                const docData = a.payload.doc.data() as DocumentData;
-                const id = a.payload.doc.id;
-                return surveyDocToModel(id, docData);
-              })
-              .filter(DataStoreService.filterAndLogError<Survey>)
-              .filter(
-                survey =>
-                  survey instanceof Survey &&
-                  (survey.generalAccess !== SurveyGeneralAccess.UNLISTED ||
-                    (survey.generalAccess === SurveyGeneralAccess.UNLISTED &&
-                      survey.ownerId === userId))
-              )
-              .map(survey => survey as Survey)
+          surveys.filter(
```

Comment:
Why not filter this in the Firebase query instead of after the fact in the client?

Reply:
because at the moment we have surveys with generalAccess = RESTRICTED and surveys without the generalAccess field. the natural and most efficient way is to add .where('generalAccess', '==', SurveyGeneralAccess.RESTRICTED) but it breaks compatibility with the old surveys.

we can improve this query once we have aligned all the old surveys setting them as RESTRICTED. I can open a new issue about it, because we can do this after the PUBLIC SURVEY feature is in production. wdyt? @gino-m